vassal-app/src/main/java/VASSAL/build/module/Console.java

```diff
@@ -0,0 +1,260 @@
+/*
+ *
+ * Copyright (c) 2020 by Brian Reynolds, VASSAL
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License (LGPL) as published by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, copies are available
+ * at http://www.opensource.org.
+ */
+package VASSAL.build.module;
+
+import VASSAL.Info;
+import VASSAL.build.GameModule;
+import VASSAL.build.module.properties.MutableProperty;
+import VASSAL.command.Logger;
+import VASSAL.tools.BugUtils;
+import org.slf4j.LoggerFactory;
+
+import java.awt.Desktop;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.regex.Pattern;
+
+/**
+ * Expandable "Console" to allow entering commands into the Chatter.
+ */
+public class Console {
+  Iterator<String> tok;
+  String commandLine;
+
+  private static final org.slf4j.Logger log =
+    LoggerFactory.getLogger(Console.class);
+
+  private void show(String s) {
+    GameModule.getGameModule().warn(s);
+  }
+
+
+  private boolean matches(String s1, String s2) {
+    return matches(s1, s2, 2);
+  }
+
+  private boolean matches(String s1, String s2, int min) {
+    if (s2.isEmpty() || (s2.length() > s1.length()) || ((s2.length() < min) && (s1.length() > s2.length()))) {
+      return false;
+    }
+    if (s2.length() < min) {
+      return s1.toLowerCase().equals(s2.toLowerCase());
+    }
+    return s1.toLowerCase().substring(0, s2.length()).equals(s2.toLowerCase());
+  }
+
+  private String nextString(String def) {
+    return tok.hasNext() ? tok.next() : "";
```

Comment:
Oops. That should be `def` here, not `""`. The empty string is what you wanted the default to be. My bad.